setup_venv.ps1

```diff
@@ -78,19 +66,27 @@ $version = if($p -is [System.Management.Automation.ErrorRecord])
 }
 else
 {
-    # otherwise return as is
-    $p
+    # otherwise return complete Python list
+    $PyVer = py --list
+}
+
+# deactivate any activated venvs
+if ($PyVer -like "*venv*")
+{
+  deactivate # make sure we don't update the wrong venv
+  $PyVer = py --list # update list
```

Comment:
Does this command work on all systems? I'm traveling and won't by and to check until Wednesday

Reply:
this should work from at least Python 2.7
I didn't check further back.
I could not break it in Powershell 5 or Powershell 7 once any Version of Python is installed. And if that's not the case, you handle it further up.